draft/XXXX-upload-system/README.md

```diff
@@ -0,0 +1,126 @@
+
+```
+{
+  collection: id,
+  hidden: bool,
+}
+```
+
+The `collection` key would point to the ID of the collection to which this document refers.
+
+Other information could be added to this collection in subsequent RFCs, making this a general purpose collection metadata storage mechanism.
+
+#### Alternate Approach
+Instead of modifying the table/network listing endpoints to omit these "hidden" tables, these endpoints could instead be modified to return objects (dictionaries), which along with the table/networks names, would include if they are temporary (i.e. being actively uploaded to). The benefit to this would be that users wouldn't be able to accidentally create duplicate tables of the same name, as they could see all tables which may exist. However, this would allow for any user (who has access to the workspace) to see an in-progress table, which may contain garbage data, and is likely unvalidated. Furthermore, if an upload fails validation after the fact, it may once again disappear from the list of tables/networks, possibly causing more confusion.
+
+### System-wide Upload Collection
+To keep track of current uploads, a `_system` collection called `uploads` would be created, which would keep track of any ongoing uploads. This collection would store, among other things, the progress of the upload, the "hidden" table(s)/graph(s) to which it belongs, and the user who initiated the upload. The schema for each document in this collection would have a structure like so:
+
+```
+{
+  current: int (bytes),
+  total: int (bytes),
+  done: bool,
+  graph: bool (indicates that a graph is being uploaded),
+  workspace: string,
+  target: {
+    tables: string[],
+    graphs: string[],
+  },
+  user: user_id (sub value),
+  _id: id (arangodb, used by clients to actually send chunks)
+}
+```
+
+### Initializing an Upload
+To initialize an upload, new endpoints will be needed, which will replace the existing upload endpoints. To initialize an upload, a request of the following format is made (endpoint for each upload format specified):
+
+```
+POST /api/upload/csv/{workspace}/{table}
+POST /api/upload/d3_json/{workspace}/{graph}
+POST /api/upload/nested_json/{workspace}/{graph}
+POST /api/upload/newick/{workspace}/{graph}
+```
+
+These endpoints replaces the original upload endpoints for each data type we currently support. This returns a JSON response, containing the upload information, in the format referenced above.
+
+A new endpoint for uploading data chunks themselves will have the following format (endpoint for each upload format specified):
+
+```
+PUT /api/upload/csv/{workspace}/{table}
+PUT /api/upload/d3_json/{workspace}/{graph}
+PUT /api/upload/nested_json/{workspace}/{graph}
+PUT /api/upload/newick/{workspace}/{graph}
```

Comment:
These should actually be `POST` endpoints, but to a different target:

```
POST /api/uploads/<upload-id>/chunk
```

Since we have upload IDs that identify the workspace/table/graph combo, there's no need to repeat them here.

This is just a sketch of how I imagine these to be. We should more or less follow Girder's practice for the design of these endpoints.